docs/INTEGRATION.md

```diff
@@ -0,0 +1,789 @@
+           │
+           ▼
+┌─────────────────────┐
+│ Fault Remediation   │ Trigger external systems (your CR)
+└─────────────────────┘
+```
+
+---
+
+## 1. Is a Node Bad? Check Taints
+
+**Use taints for all scheduling and automation decisions.**
+
+Taints are the primary signal that a node has hardware issues. External systems should watch for taint presence/absence to make scheduling decisions, trigger alerts, or initiate remediation workflows.
+
+> **Note**: Taints are **optional and disabled by default**. You must configure them in fault-quarantine rulesets by uncommenting the `taint` section. NVSentinel only cordons nodes by default.
+
+### Taint Structure
+
+**Format:** User-configurable via rulesets. Common patterns:
+
+**Option 1: Component-specific (recommended)**
+```
+nvidia.com/gpu-xid-error
+nvidia.com/gpu-nvlink-error
+nvidia.com/syslog-xid-error
+```
+
+**Option 2: Hierarchical (proposed pattern)**
+```
+gpu.health/memory-error
+nvlink.health/link-down
+nvswitch.health/fatal-error
+```
+
+### Default Taint Examples (From Test Suite)
+
+NVSentinel's test suite demonstrates these taint configurations:
+
+| Taint Key                       | Value  | Effect       | Use Case                    |
+|---------------------------------|--------|--------------|
+| `nvidia.com/gpu-xid-error`      | `true` | `NoSchedule` | GPU XID critical errors     |
+| `nvidia.com/gpu-nvlink-error`   | `true` | `NoSchedule` | NVLink connection failures  |
+| `nvidia.com/syslog-xid-error`   | `true` | `NoSchedule` | Syslog-detected XID errors  |
+| `nvidia.com/gpu-error`          | `true` | `NoSchedule` | Generic GPU hardware errors |
+
+You can configure any taint keys/values in your rulesets based on your needs.
+
+### Taint Effect Guidelines
+
+| Effect              | Use Case                         | Impact                                          |
+|---------------------|----------------------------------|-------------------------------------------------|
+| `NoSchedule`        | Fatal errors requiring remediation | New pods without toleration won't be scheduled |
+| `PreferNoSchedule`  | Degraded state or warnings       | Scheduler tries to avoid but will schedule if necessary |
+| `NoExecute`         | Immediate evacuation needed      | Existing pods without toleration are evicted (rarely used) |
+
+### Configuring Taints
+
+Taints are defined in Fault Quarantine rulesets. Here's an example showing how to enable taints:
+
+```yaml
+# distros/kubernetes/nvsentinel/charts/fault-quarantine/values.yaml
+rulesets:
+  - version: "1"
+    name: "GPU XID Critical Errors"
+    priority: 100
+    match:
+      any:
+        - kind: "HealthEvent"
+          expression: 'event.checkName == "GpuXidError" && event.isFatal == true'
+    # Uncomment to enable tainting:
+    taint:
+      key: "nvidia.com/gpu-xid-error"  # Choose your own key format
+      value: "true"                    # Or use "fatal", "degraded", etc.
+      effect: "NoSchedule"
```

Comment:
since we say uncomment, shouldn't the lines be commented?

```suggestion
    #taint:
    #  key: "nvidia.com/gpu-xid-error"  # Choose your own key format
    # value: "true"                    # Or use "fatal", "degraded", etc.
    # effect: "NoSchedule"
```